### PR TITLE
Reduce CUDA kernel launch in BN

### PR DESCRIPTION
**Pull request by okuta**
Currently `MultiNodeBatchNormalizationFunction` calls 4 multiply and 2 add operations. 
This PR unifies it kernel launch.

### CODE REVIEW

File: chainermn/functions/batch_normalization.py [hunk 1]
```diff
@@ -160,16 +160,28 @@ def forward(self, inputs):
...
+                temp_ar *= (1 - self.decay)
+                self.running_mean += temp_ar
+                del temp_ar
+                self.running_var *= self.decay
+                temp_ar = xp.array(var)
+                temp_ar *= (1 - self.decay) * adjust
+                self.running_var += temp_ar
+                del temp_ar
+            else:
+                cuda.elementwise(
```
Comment (takerum) [inline chainermn/functions/batch_normalization.py]: syntax doesn't seem to be correct. maybe these lines should be changed to:
```
cuda.elementwise(
                    'T mean, T var, T decay, T adjust',
                    'T r_mean, T r_var',
                    '''
                    r_mean = r_mean * decay + mean * (1 - decay);
                    r_var = r_var * decay + var * (1 - decay) * adjust;
                    ''',
                    'update_mean_var'
```
?

Comment (okuta) [reply chainermn/functions/batch_normalization.py]: Thanks!

Comment (okuta) [reply chainermn/functions/batch_normalization.py]: I fixed.

Comment (takerum) [reply chainermn/functions/batch_normalization.py]: the error is not completely fixed. The comma after `T r_var` should be removed.
https://github.com/chainer/chainermn/pull/276/commits/8d37ea636aa9f5e37a029c39407e0efe6cdf3436#diff-3c18be054f7dc155b2c3b9bda576278bL177

Comment (okuta) [reply chainermn/functions/batch_normalization.py]: ohh, thanks.

**Comment by keisukefukuda** [comment]
Replaced by #282

**Pull request closed without merging**